vm/parser.ts

```diff
@@ -711,6 +861,17 @@ module J2ME {
       }
     }
 
+    getClassNameSlow(): string {
+      return this._name || (this._name = ByteStream.readString(this.utf8Name));
+    }
```

Comment:
> The memoizer version falls behind here because it allocates a lot of temporary objects. Here is a workaround for that: http://jsfiddle.net/dt4z21af/6/. This closes the gap but falls behind when the number of accesses is low and the number of objects is high.

Indeed, it looks like *Object.defineProperty* is responsible for the cost. Replacing it with a private *this._x* member makes the prototype getter (no longer accurate to call it a memoizer) twice as fast again:

http://jsfiddle.net/dt4z21af/7/

Things change somewhat if the value is computed (as it will be in our use case). Then, the prototype getter is still faster, but the numbers are closer, and occasionally flip:

http://jsfiddle.net/dt4z21af/8/

Which still litigates for the prototype getter.